docs/advanced-solidus/state-machines.mdx

```diff
@@ -1,211 +1,88 @@
-between state machine states are good candidates to become hotspots where tangential logic is
-triggered.
-
-For instance, you might want to update your ERP or send an SMS when a payment is marked as
-completed. The cleaner way to do that is to publish an event when that happens and then subscribe to
-it. First, you need to override the `#complete` event method on `Spree::Payment` (see
-the [overrides section](../customization/customizing-the-core.mdx#using-overrides) for the required
-setup code):
-
-```ruby title="app/overrides/my_store/publish_payment_completed.rb"
-# frozen_string_literal: true
-
-module MyStore
-  module PublishPaymentCompleted
-    def complete
-      super.tap do |result|
-        Spree::Bus.publish(:payment_completed, payment: self) if result
-      end
-    end
-
-    ::Spree::Payment.prepend self
-  end
-end
-```
-
-Then, after [registering the new event](../customization/subscribing-to-events.mdx#custom-events),
-you can [create a subscriber](../customization/subscribing-to-events.mdx#subscription-to-events) for
-it.
-
-:::caution
-
-Why not register a new `after_transition` hook instead of overriding the event method (
-i..e, `#complete` in the example above)?
-
-State machine transitions (including their `after_` hooks) are wrapped within a database
-transaction. By overriding the method instead of using `after_transition`, we publish the bus event
-only after the transaction has been committed to the DB.
-
-Event subscribers should **always** be decoupled from the main transaction, so there's no point in
-blocking database access until your subscribers have finished running. In fact, it's a bad practice:
-a failed subscriber will roll back the whole DB transaction and potentially leave your system in an
-inconsistent state.
-
-:::
-
-:::info
-
-Ideally, Solidus would publish events for every state machine transition out of the box. Our event
-bus is fairly new and we're still working on it, but we'll get there eventually! In the meantime,
-you can check`Spree::Bus.registered_events` for the complete list of events that are already
-published.
+It's better to be conservative when customizing state machines. Try to apply the smallest possible
+set of changes, and if possible, avoid changing the defined states. You're dealing with the core of
+the domain model: large changes could branch out in unanticipated ways!
 
 :::
 
-### Using custom state machines
-
-If needed, you can flat out replace state machines with your custom implementation. This can be done
-through the `state_machines` option in `config/initializers/spree.rb`.
-
-For instance, if you want to replace the payment state machine, you can do it like this:
-
-```ruby title="lib/my_store/state_machines/payment.rb"
-# frozen_string_literal: true
+## How-to guides
 
-module MyStore
-  class StateMachines
-    module Payment
-      extend ActiveSupport::Concern
+- [How to customize core behavior in state machines][how-to-customize-core-behavior]
```

Comment:
These three how-to names are a bit generic, and more a technical description of what they are doing than what people actually search for: I have the feeling people will need to open them and read a bit to understand what we are talking about. Some proposals might be:

`How to customize core behavior in state machines` -> `How to customize existing state machines` or `How to add transitions to existing state machines`
`How to use custom state machines` -> `How to replace an existing state machine`
`How to add orthogonal behavior` -> 🤷‍♂️ I don't know, it's easy to link this too much with events but I think that there are other scenarios as well.

Reply:
-How to customize core behavior in state machines -> How to customize existing state machines ✅ 
-How to use custom state machines -> How to replace an existing state machine ✅ 
- How to add orthogonal behavior -> How to add orthogonal behavior: publishing events ✅ (I've gone with the same solution [we reached for skipping RMAs in returns](https://github.com/solidusio/edgeguides/pull/61#discussion_r959560113))